packages/ui/src/components/Button/__stories__/Extend.stories.tsx

```diff
@@ -1,11 +1,75 @@
-import { Template } from './Template.stories'
+import type { ComponentStory } from '@storybook/react'
+import Button from '..'
+import Stack from '../../Stack'
 
-export const Extend = Template.bind({})
+const Buttons: ComponentStory<typeof Button> = ({ ...props }) => (
+  <>
+    <div>
+      <Button icon="plus" iconSize={24} extend size="large" {...props}>
+        Normal Button
+      </Button>
+    </div>
+    <div>
+      <Button icon="plus" iconSize={24} extend size="medium" {...props}>
+        Normal Button
+      </Button>
+    </div>
+    <div>
+      <Button icon="plus" iconSize={24} extend size="small" {...props}>
+        Normal Button
+      </Button>
+    </div>
+    <div>
+      <Button icon="plus" iconSize={24} extend size="xsmall" {...props}>
+        Normal Button
+      </Button>
+    </div>
+    <div>
+      <Button icon="plus" iconSize={24} extend size="xxsmall" {...props}>
+        Normal Button
+      </Button>
+    </div>
+    <div>
+      <Button
+        icon="plus"
+        iconSize={24}
+        extend
+        as="a"
+        href="https://scaleway.com"
+        {...props}
+      >
+        Link Button
+      </Button>
+    </div>
+    <div>
+      <Button
+        icon="plus"
+        iconSize={24}
+        extend
+        as="a"
+        href="https://scaleway.com"
+        size="small"
+        {...props}
+      >
+        Link Button
+      </Button>
+    </div>
+  </>
+)
 
-Extend.args = {
-  icon: 'plus',
-  extend: true,
-}
+export const Extend: ComponentStory<typeof Button> = ({ ...props }) => (
+  <Stack gap={1} style={{ width: '100%' }}>
+    <Stack gap={1}>
+      <Buttons {...props} />
+    </Stack>
+    <Stack gap={1} alignItems="center">
+      <Buttons {...props} />
+    </Stack>
+    <Stack gap={1} alignItems="end">
+      <Buttons {...props} />
+    </Stack>
+  </Stack>
+)
```

Comment:
I'm afraid you cannot use this method as example in storybook, it doesn't show properties on `Button` component itself but on the `Buttons` you created.

<img width="1001" alt="Screenshot 2023-01-03 at 10 34 22" src="https://user-images.githubusercontent.com/15812968/210331554-b31f1cf2-4ccd-4304-8114-cb08af08a38d.png">

Instead I will recommend something shorter but more clear:
```ts
export const Extend: ComponentStory<typeof Button> = ({ ...props }) => (
  <Stack gap={1} style={{ width: '100%' }}>
    <Stack gap={1}>
      <Button icon="plus" iconSize={24} extend size="medium" {...props}>
        Normal Button
      </Button>
    </Stack>
    <Stack gap={1} alignItems="center">
      <Button icon="plus" iconSize={24} extend size="medium" {...props}>
        Normal Button
      </Button>
    </Stack>
    <Stack gap={1} alignItems="end">
      <Button icon="plus" iconSize={24} extend size="medium" {...props}>
        Normal Button
      </Button>
    </Stack>
  </Stack>
)
```

This way the example will be filly shown in code snippet, it is also shorter so easier to read.